### PR TITLE
[SVCS-939] Upgrade WB base python to 3.6

### PR DESCRIPTION
**Pull request by ExProbitasFiducia**
This PR is pointing at the next branch.

Can't upgrade to 3.7 yet because our old version of aiohttp is blocking
it. async becomes a reserved word, but the old version of aiohttp uses
that.

This pr will require another pr on osf.io to update `docker-compose`

## Ticket

https://openscience.atlassian.net/browse/SVCS-939

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+version: "3"
```
Comment (cslzchen) [inline docker-compose.yml]: * Personally, I still prefer this to be a local file for developers who need it.
* In addition, this PR shouldn't include this change since it is not related with Python upgrade.

File: requirements.txt [hunk 2]
```diff
@@ -11,7 +11,7 @@ pyjwe==1.0.0
 pyjwt==1.4.0
 python-dateutil==2.5.3
 pytz==2017.2
-raven==5.27.0
+raven==6.9.0
```
Comment (cslzchen) [inline requirements.txt]: Is the `raven` upgrade blocking `python` upgrade? I can't find docs supporting this claim. I think we can have a separate PR for upgrading `raven.`

File: waterbutler/core/utils.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def as_task(func):
 
     @functools.wraps(func)
     def wrapped(*args, **kwargs):
-        return asyncio.async(func(*args, **kwargs))
+        return asyncio.ensure_future(func(*args, **kwargs))
```
Comment (cslzchen) [inline waterbutler/core/utils.py]: 👍 fyi, here is the ref: https://docs.python.org/3.6/library/asyncio-task.html#asyncio.ensure_future

> `asyncio.async(coro_or_future, *, loop=None)`
A deprecated alias to ensure_future().
Deprecated since version 3.4.4.

**Comment by cslzchen** [comment]
In addition, please create the a ticket or link one. Thanks.

**Comment by felliott** [comment]
Done in d03a84c7046baed176. Closing.
